src/App.tsx

```diff
@@ -1,73 +1,65 @@
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
+import { Autocomplete } from './components/Autocomplete';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const { name, born, died } = selectedPerson || {};
+  const [delay] = useState(300);
+  const [query, setQuery] = useState('');
+  const [focus, setFocus] = useState(false);
+  const timerId = useRef(0);
+
+  function saveQuery(newQuery: string) {
+    window.clearTimeout(timerId.current);
+
+    timerId.current = window.setTimeout(() => {
+      setQuery(newQuery.trim());
+    }, delay);
+  }
+
+  window.addEventListener('click', () => {
+    setFocus(false);
+  });
```

Comment:
Using `window.addEventListener` directly in the component can lead to memory leaks because the event listener is not removed when the component unmounts. Consider using a `useEffect` hook to add and clean up the event listener.